### DISCUSSION TITLE
Showcase trending/popular packages on SPI

### DISCUSSION THREAD

**Discussion by jeroenleenarts** [SwiftPackageIndex/SwiftPackageIndex-Server#1807] [Ideas]
I know there is some work going into scoring/assigning a qualitative value to packages.

But what about trending packages on SPI? It would be great to get some insight into what packages are being visited/looked up most frequently.

A dampening algorithm could be implemented suppressing highly popular subset of packages topping the charts. Something that takes tanking of previous weeks/days into account.

Score top of chart should be strictly numbers based. Scoring high multiple weeks in a row should be progressively harder. Once a package drops off a cool down can trigger a reset of the dampening inputs.
It could even be based on comparing traffic compared over two weeks. If one week the traffic is all of a sudden doubled. Hey. That is trending.

The ranging source doesn't have to be perfect I think. Something like logging async to elasticseach should be all that's needed. Log a timestamp, path when it is the package page, and your are done. The ranking lists including dampening should be a matter of writing the right elastic queries right?

**Comment by daveverwer** [comment]
Yes, one advantage that package index type sites for other languages have is that they're usually tied into the tool that does the package download so they can get a sense of how many projects are actually importing a library. I'm quite sure we can't do that right now, and we'd want to think really carefully about collecting that kind of data even if it were possible. It may be something that the package registry service could provide.

We do have package page analytics but right now they are very much dominated by the packages I link to in iOS Dev Weekly.  Currently, there are only two packages in the top ten from the last 30 days that I didn't link to. 😬

<img width="300" alt="Screenshot 2022-06-14 at 14 07 17@2x" src="https://user-images.githubusercontent.com/5180/173584446-6838ac3f-d89a-4d99-a667-bdf9e3f40f95.png">
 
So, while it'd be possible to either instrument this kind of statistic outside of analytics (because the analytics data is siloed right now) or do some kind of data processing to get it into a place where we could use it, I don't think now is the time. The site continues to increase in popularity every month, but my influence is still way too strong on it and that's not something I feel comfortable using in this way.

If it became standard to link to SPI over GitHub, that would change quickly, but that's a hard hill to climb and we're barely out of the gift shop at base camp 😂

**Reply by jeroenleenarts** [reply]
Some solid thoughts there.

I am mainly looking for a fun way to surface interesting Swift Packages. Maybe a randomized list of packages in good standing (3 stars or more based on #1639) would be an alternative?
